### PR TITLE
New car example UI and pagination feature

### PR DESCRIPTION
**Pull request by Wouter125**
The car example UI has been reworked to the new design. Alongside that we also incorporated some new features;

- [x] Endless scroll pagination support
- [x] Improved search support
- [x] Sticky make headers 

We are planning on writing an article around it as well on Medium.

### CODE REVIEW

File: car/index.js [hunk 1]
```diff
@@ -14,26 +25,188 @@ const headers = {
...
 
+// Keeps track of the active search keyword. This way we always fetch the right results when using pagination.
+let searchKeyword = '';
+
+// Keeps track of which page of cars we are currently on.
+let currentPage = 0;
+
+/**
+ * The function that handles all our GraphQL networking alongside it's parameters.
+ * @param {Object} - Object that manages the page, size and search to be send towards our request
```
Comment (timkuilman) [inline car/index.js:44]: ``` @param { page: number, size: number, search: string } - Object that manages the page, size and search to be send towards our request```

You can define the keys in the objects to be more specific about the input object. Can you do this as well for the other functions?

File: car/index.js [hunk 2]
```diff
@@ -14,26 +25,188 @@ const headers = {
...
+ */
+const fetchCars = ({ page, size = 10, search = '' }) => {
+  client
+    .query(getCarList, { page: page, size: size, search: search })
```
Comment (timkuilman) [inline car/index.js:48]: .query(getCarList, { page, size, search })

**Review by timkuilman — CHANGES_REQUESTED** [review]
Clear example! The only part that I found harder to follow and could use a more comments is the grouping part. I don't think I would have gone for a Map here. I think the reduce method would make more sense here. 

Currently the `groupCars` function not only groups the the cars but also calls the `renderCarList`. Can we change the first function to return the groupedCars (I think you should use reduce) and move calling the `renderCarList to the eventListener?

File: car/index.js (hunk 1 above)
Comment (Wouter125) [reply car/index.js:44]: I'm fine with it but then I'll probably have to create an interface / type for this one;

* @param { Object } cars - A list of cars coming from the Chargetrip API

Not sure whether that's an overkill for an example.

Comment (timkuilman) [reply car/index.js:44]: Creating the actual TS interfaces is not necessary since the examples are not in TS.

**Review by timkuilman — APPROVED** [review]

**Pull request merged**